src/palace.pyx

```diff
@@ -689,14 +686,12 @@ cdef class Source:
 
     Parameters
     ----------
-    context : Optional[Context]
+    context : Context
         The context from which the source is to be created.
-        If it is `None`, the object is left uninitialized.
     """
     cdef alure.Source impl
 
-    def __init__(self, context: Optional[Context]) -> None:
-        if context is None: return
+    def __init__(self, context: Context) -> None:
         self.impl = (<Context> context).impl.create_source()
```

Comment:
Notice the unnecessary casting here.

Reply:
so we dont cast it to None?

Reply:
What, no, I mean `context` needn't be casted (`(<Context> context)`).

Reply:
Drop the parentheses please.

Reply:
Gentle ping (both cases), other wise LGTM.